pilot/cmd/sidecar-injector/main.go

```diff
@@ -95,22 +109,66 @@ var (
 	}
 )
 
+func patchCert() error {
+	const retryTimes = 6 // Try for one minute.
+	client, err := createClientset(flags.kubeConfigFile)
+	if err != nil {
+		return err
+	}
+	caCertPem, err := ioutil.ReadFile(flags.caCertFile)
+	if err != nil {
+		return err
+	}
+	i := 0
+	for i < retryTimes {
+		err = util.PatchMutatingWebhookConfig(client.AdmissionregistrationV1beta1().MutatingWebhookConfigurations(),
+			flags.webhookConfigName, flags.webhookName, caCertPem)
+		if err == nil {
+			return nil
+		}
+		log.Errorf("Register webhook failed: %s. Retrying...", err)
+		time.Sleep(time.Second * 10)
+	}
+	return err
+}
+
+func createClientset(kubeConfigFile string) (*kubernetes.Clientset, error) {
+	var err error
+	var c *rest.Config
+	if kubeConfigFile != "" {
+		c, err = clientcmd.BuildConfigFromFlags("", kubeConfigFile)
+	} else {
+		c, err = rest.InClusterConfig()
+	}
+	if err != nil {
+		return nil, err
+	}
+	return kubernetes.NewForConfig(c)
+}
+
 func init() {
 	rootCmd.PersistentFlags().StringVar(&flags.meshconfig, "meshConfig", "/etc/istio/config/mesh",
 		"File containing the Istio mesh configuration")
 	rootCmd.PersistentFlags().StringVar(&flags.injectConfigFile, "injectConfig", "/etc/istio/inject/config",
 		"File containing the Istio sidecar injection configuration and template")
-	rootCmd.PersistentFlags().StringVar(&flags.certFile, "tlsCertFile", "/etc/istio/certs/cert.pem",
+	rootCmd.PersistentFlags().StringVar(&flags.certFile, "tlsCertFile", "/etc/istio/certs/cert-chain.pem",
 		"File containing the x509 Certificate for HTTPS.")
 	rootCmd.PersistentFlags().StringVar(&flags.privateKeyFile, "tlsKeyFile", "/etc/istio/certs/key.pem",
 		"File containing the x509 private key matching --tlsCertFile.")
+	rootCmd.PersistentFlags().StringVar(&flags.caCertFile, "caCertFile", "/etc/istio/certs/root-cert.pem",
+		"File containing the x509 Certificate for HTTPS.")
 	rootCmd.PersistentFlags().IntVar(&flags.port, "port", 443, "Webhook port")
 
 	rootCmd.PersistentFlags().DurationVar(&flags.healthCheckInterval, "healthCheckInterval", 0,
 		"Configure how frequently the health check file specified by --healhCheckFile should be updated")
 	rootCmd.PersistentFlags().StringVar(&flags.healthCheckFile, "healthCheckFile", "",
 		"File that should be periodically updated if health checking is enabled")
-
+	rootCmd.PersistentFlags().StringVar(&flags.kubeConfigFile, "kubeConfig", "",
```

Comment:
s/`kubeConfig`/`kubeconfig` to match other flag names (see below).

```bash
$ grep -irn kubeconfig |grep PersistentFlags
pilot/cmd/pilot-discovery/main.go:85:   discoveryCmd.PersistentFlags().StringVar(&serverArgs.Config.KubeConfig, "kubeconfig", "",
mixer/cmd/mixs/cmd/validator.go:77:     validatorCmd.PersistentFlags().StringVar(&kubeconfig, "kubeconfig", "", "Use a Kubernetes configuration file instead of in-cluster configuration")
broker/cmd/brks/cmd/server.go:45:       serverCmd.PersistentFlags().StringVar(&sa.kubeconfig, "kubeconfig", "",
istioctl/cmd/istioctl/main.go:514:      rootCmd.PersistentFlags().StringVarP(&kubeconfig, "kubeconfig", "c", defaultKubeConfigText,
```

re: environment variable vs. flag, it looks like Mixer, Pilot, Broker, and Istioctl all use `kubeconfig` and, in two non-test cases, use `KUBECONFIG` as a fall back (see below). Flags are more discoverable (i.e. they show up in auto-generated docs.

```bash
pilot/pkg/serviceregistry/kube/client.go:39:            kubeconfig = os.Getenv("KUBECONFIG")
tests/k8s/helper.go:26: kubeconfig := os.Getenv("KUBECONFIG")
tests/e2e/tests/pilot/util/config.go:66:                KubeConfig:            os.Getenv("KUBECONFIG"),
tests/e2e/tests/pilot/controller_test.go:34:    cl, err := crd.NewClient(os.Getenv("KUBECONFIG"), desc, "")
tests/e2e/tests/pilot/controller_test.go:53:    _, client, err := kube.CreateInterface(os.Getenv("KUBECONFIG"))
istioctl/cmd/istioctl/main.go:504:              if v := os.Getenv("KUBECONFIG"); v != "" {
```